redis/conn.go

```diff
@@ -137,6 +148,21 @@ func Dial(network, address string, options ...DialOption) (Conn, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if do.dialTLS {
+		tlsConfig := do.tlsConfig
+		if tlsConfig == nil {
+			// https://golang.org/pkg/crypto/tls/#Client
+			// At this point we don't know the ServerName.
+			tlsConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
I guess I could also set a different dialer instead that is a wrapper around tls.Dial like so:
```go
DialNetDial(func(network, addr string) (net.Conn, error) {
  tls.Dial(network, addr, do.tlsConfig)
})
```

That actually may make more sense.